zeebe/engine/src/main/java/io/camunda/zeebe/engine/metrics/DistributionMetrics.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.metrics;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Ideas:
+ *
+ * <p>- TimeGauge to track the age of the current/oldest distribution? or last latency to another
+ * partition? Probably better a histogram /timer
+ *
+ * <p>TimeGauge is probably better for onRecovery Tracking???
+ *
+ * <p>- ... - ... - ...
+ */
+public final class DistributionMetrics {
+
+  private final MeterRegistry meterRegistry;
+  private final Map<Integer, PartitionDistributionMetrics> partitionMetrics =
+      new ConcurrentHashMap<>(); // TODO check if concurrent map is needed
+
+  public DistributionMetrics(final MeterRegistry meterRegistry) {
+    this.meterRegistry = meterRegistry;
+
+    // TODO: setup micrometer metrics once (performance)
+  }
+
+  /**
+   * Resets the metrics (e.g. in case of recovery to remain in sync with the state). Be careful
+   * about race conditions and call this only from the stream processing actor
+   */
+  public void reset(final int counter) {}
+
+  public void addDistribution(final long distributionKey) {
```

Comment:
:question: Why always pass the keys? We wouldn't actually add this as a label/tag, since the cardinality is very very high, right?

Reply:
We possibly need this or something similar for the timer/latency metrics (If not I will remove them with that PR)